rpc/tests/p2p.rs

```diff
@@ -0,0 +1,273 @@
+use crate::utils::client::{new_test_client, AuthLevel};
+use celestia_rpc::prelude::*;
+use celestia_types::p2p;
+use libp2p::{identity, PeerId};
+use tokio::time::{sleep, Duration};
+
+pub mod utils;
+
+async fn rpc_call_delay() {
+    // delay for RPC calls like connect/close to let node finish the operation before we query it
+    // again. Below 150 ms I start getting intermittent failures.
+    sleep(Duration::from_millis(150)).await;
+}
+
+#[tokio::test]
+async fn info_test() {
+    let client = new_test_client(AuthLevel::Admin).await.unwrap();
+    client.p2p_info().await.expect("Failed to get node info");
+}
+
+#[tokio::test]
+async fn add_remove_peer_test() {
+    // add and then remove a peer, testing outputs from `p2p.Peers` and `p2p.Connectedness`
+    let (addr_info, task) = utils::tiny_node::start_tiny_node()
+        .await
+        .expect("failed to spin up second node");
+    let client = new_test_client(AuthLevel::Admin).await.unwrap();
+
+    let initial_peers = client
+        .p2p_peers()
+        .await
+        .expect("failed to get initial peer list");
+    assert!(!initial_peers.contains(&addr_info.id));
+
+    let connected_to_peer = client
+        .p2p_connectedness(&addr_info.id)
+        .await
+        .expect("failed to check initial connection to peer");
+    assert_eq!(connected_to_peer, p2p::Connectedness::NotConnected);
+
+    client
+        .p2p_connect(&addr_info)
+        .await
+        .expect("request to connect to second node failed");
+    rpc_call_delay().await;
+
+    let peers = client
+        .p2p_peers()
+        .await
+        .expect("failed to get peer list after connect request");
+    assert!(peers.contains(&addr_info.id));
+
+    let connected_to_peer = client
+        .p2p_connectedness(&addr_info.id)
+        .await
+        .expect("failed to check connection to peer after connect request");
+    assert_eq!(connected_to_peer, p2p::Connectedness::Connected);
+
+    client
+        .p2p_close_peer(&addr_info.id)
+        .await
+        .expect("Failed to close peer");
+    rpc_call_delay().await;
+
+    let final_peers = client
+        .p2p_peers()
+        .await
+        .expect("failed to get peer list after close peer request");
+    assert!(!final_peers.contains(&addr_info.id));
+
+    task.abort();
```

Comment:
cleanup aborts inside `tokio::test` macro are redundant as this is handled by the macro, please apply to other too

nevertheless, those are particularly a wrong pattern, you should create some new struct with `Drop` for such cleanup as `drop` fires on panics, where this does not. E.g. if someone would create a code that fails some of the asserts, he won't have the process cleaned up

Reply:
> cleanup aborts inside tokio::test macro are redundant as this is handled by the macro

not sure that's perfectly true tho

Reply:
yeah, so that wasn't true. So non-awaited tasks that keep running after main thread finishes are just killed

Reply:
this seems to be also an issue for our current node architecture, will try to fix that with the traits PR

Reply:
https://docs.rs/tokio/latest/tokio/attr.test.html#current-thread-runtime each test is run in it's own runtime and I see that tasks spawned by test stop when test exists, so removing manual abort.